### PR TITLE
Refactor for builder changes and Individual Datasets

### PR DESCRIPTION
**Pull request by JLoveUOA**
Two major changes in this PR.

### Refactored based on builder changes

- Encrypted graph is now expected to be written within the RO-Crate graph
- Sensitive data now generates recipients entities based on public keys supplied.
- Bulk encryption of output archive is now supported via the --bulk-encrypt command
- Metadata holds schema url information
- groups are represented as ACLS
- New Data types are created in the RO-Crate
    - Organizations
    - facilities
    - group (for both ACLS and facility manager groups)
    - ACLS as Digital Document Permissions

### Print Lab Extractor now Produces individual dataset Crates

- Datasets now collect all sub-files and directories ONLY If 'crate children' is specified in the input worksheet
    - The --duplicate-directory parameter will also cause each dataset to fully duplicate the crate source.
- the --split datasets parameter will now produce an individual dataset RO-Crate for each dataset in the spreadsheet
- --duplicate-directory and --split-datasets are not recommend to be used together (this duplicates your data N times where N is your number of datasets)
- Datasets hold an individual copy of all relevant metatada 

## Added tests for all print lab data extraction and generic metadata extraction

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -7,30 +7,28 @@ readme = "README.md"
...
 backoff = "^2.2.1"
 requests = "^2.31.0"
-build = "^1.1.1"
 pydantic-settings = "^2.2.1"
-ro-crate-py = {git = "https://github.com/UoA-eResearch/ro-crate-py.git",  branch = "encrypted-metadata"}
 mytardis-rocrate-builder = {git = "https://github.com/JLoveUOA/mytardis_rocrate_builder"}
+ro-crate-py = {git = "https://github.com/UoA-eResearch/ro-crate-py.git",  branch = "encrypted-metadata"}
+openpyxl = "^3.1.5"
+click = "^8.1.7"
+faker = "^27.4.0"
```
Comment (ChrisKSeal) [inline pyproject.toml]: Do Faker, FactoryBoy and Hypothesis want to sit in general dependencies or should these be dev dependencies? I.e. do we expect our users to run the tests?

Just a thought

File: src/ingestion_targets/print_lab_genomics/__init__.py
Comment (ChrisKSeal) [inline src/ingestion_targets/print_lab_genomics/__init__.py:1]: Should this be in a separate .env file? or in the conftest.py?

Comment (JLoveUOA) [reply src/ingestion_targets/print_lab_genomics/__init__.py:1]: There is a value in the ENV file that updates this with a new pubkey this one is just hard-coded in by default.
I should definitely update the email to a dummy value.

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,905 @@
...
+        "Project PI": ["pmgc006", "gbak002"],
+        "Ethics Approval ID": ["aaaa", None],
+        "Ethics Approval Designation": ["www.google.com", None],
+        "description": ["text", "this is a project"],
+    }
+    return pd.DataFrame(projects_data)
+
+
+def fake_upi(faker: Faker) -> str:
+    return faker.bothify(text="???###", letters="abcdefghijklmnopqrstuvwxyz")
```
Comment (ChrisKSeal) [inline tests/conftest.py]: Is this missing a character? Should be ????### IIRC :)

File: tests/examples_for_test/print_lab_test/.~lock.sampledata.xlsx#
Comment (ChrisKSeal) [inline tests/examples_for_test/print_lab_test/.~lock.sampledata.xlsx#:1]: I'm guessing this is included by mistake :)

File: tests/test_apis.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+from src.ingestion_targets.print_lab_genomics.ICD11_API_agent import ICD11ApiAgent
+from src.ingestion_targets.print_lab_genomics.print_crate_dataclasses import (
+    MedicalCondition,
+)
+from src.mt_api.api_consts import CONNECTION__HOSTNAME
+from src.mt_api.apiconfigs import AuthConfig, MyTardisRestAgent
+
+
+@responses.activate
+def test_mytardis_rest_agent_get(
```
Comment (ChrisKSeal) [inline tests/test_apis.py:22]: Not sure what this test is aiming to do. The response has been mocked so asserting that the status is 200 is not useful here. It does demonstrate that a get request has been made :)

File: tests/test_apis.py [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    responses.add(
+        responses.GET,
+        url,
+        status=200,
+        json=test_metadata_response,
+    )
+
+    response = mt_rest_agent.no_auth_request("GET", url)
+    do_not_use_auth.assert_not_called()
+    assert isinstance(response.json(), dict)
```
Comment (ChrisKSeal) [inline tests/test_apis.py]: These three tests are mocked so asserting them is surplus to requirements I think.

**Review by ChrisKSeal — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (JLoveUOA) [reply pyproject.toml]: Moved them to a test group, I don't think I even ended up needing hypothesis and factory boy so will remove those

File: tests/test_apis.py (hunk 3 above)
Comment (JLoveUOA) [reply tests/test_apis.py:22]: Basically all that was for, that a request happens at all. Added a few tests for bad requests to make it more complete.

File: tests/test_apis.py (hunk 4 above)
Comment (JLoveUOA) [reply tests/test_apis.py]: reduced just down to checking the auth is not used

File: tests/examples_for_test/print_lab_test/.~lock.sampledata.xlsx#
Comment (JLoveUOA) [reply tests/examples_for_test/print_lab_test/.~lock.sampledata.xlsx#:1]: yes removed :)

**Pull request merged**